plugin/kubernetes/controller.go

```diff
@@ -395,3 +396,11 @@ func (dns *dnsControl) GetNodeByName(name string) (*api.Node, error) {
 	}
 	return v1node, nil
 }
+
+func (dns *dnsControl) GetNamespaceByName(name string) (*api.Namespace, error) {
+	v1ns, err := dns.client.Namespaces().Get(name, meta.GetOptions{})
```

Comment:
I think this is a round trip to the k8s API (slow).  We should probably maintain a cache of namespaces instead, like we do for services, endpoints, and pods in `pods verified` mode.

Reply:
It is, as is `GetNodeByName` (which is only used by federation?). The use case here is also small For the time being update the comment and saying this is slow because of the round trip should be good enough? I.e. it is only called on "wildcard" requests.

Reply:
Federation in k8s is in limbo. Stuck in beta, it's likely to be scrapped and replaced with something entirely different. So, I would not be concerned for performance there.

Wildcards requests are non-standard, so I'm sure they are practically a non-existent.
I don't know how common bare requests are (e.g. `ns1.svc.cluster.local`), 

So as long as bare namespace requests are uncommon, then it's OK to be slow here.

Reply:
Should add namespace exposure cases to unit tests.